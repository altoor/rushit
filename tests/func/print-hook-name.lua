client_socket( function () print('TRACE:client_socket') end )
client_sendmsg( function () print('TRACE:client_sendmsg'); return 0 end )
client_recvmsg( function () print('TRACE:client_recvmsg'); return 0 end )
client_recverr( function () print('TRACE:client_recverr'); return 0 end )
client_close( function () print('TRACE:client_close') end )

server_socket( function () print('TRACE:server_socket') end )
server_sendmsg( function () print('TRACE:server_sendmsg'); return 0 end )
server_recvmsg( function () print('TRACE:server_recvmsg'); return 0 end )
server_recverr( function () print('TRACE:server_recverr'); return 0 end )
server_close( function () print('TRACE:server_close') end )
