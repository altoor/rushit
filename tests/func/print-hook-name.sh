#!/bin/bash

set -o errexit

basedir=$(dirname "$0")
script=$basedir/print-hook-name.lua
server_out=$(mktemp)
client_out=$(mktemp)

./dummy_test --script $script | grep '^TRACE:'> $server_out &
server_pid=$!

./dummy_test --script $script --client | grep '^TRACE:' > $client_out &
client_pid=$!

wait $client_pid
wait $server_pid

cmp $basedir/print-hook-name.server.out $server_out
cmp $basedir/print-hook-name.client.out $client_out

rm $server_out $client_out
