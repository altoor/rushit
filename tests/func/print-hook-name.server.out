TRACE:server_socket
TRACE:server_recvmsg
TRACE:server_sendmsg
TRACE:server_recverr
TRACE:server_close
